src/main/java/org/fenixedu/academic/domain/degreeStructure/CycleType.java

```diff
@@ -157,4 +159,14 @@ public boolean hasPrevious() {
         return getPrevious() != null;
     }
 
+    public LocalizedString getDescriptionI18N() {
```

Comment:
Again, and for the same reasons, getDescriptionI18N should be the source:

```java
    public LocalizedString getDescriptionI18N() {
        return BundleUtil.getLocalizedString(Bundle.ENUMERATION, getQualifiedName());
    }
```

and ```getDescription(Locale)``` should just invoke this with ```.getContent(locale)``` after